docs/training/apis-howto/api-core-ug.rst

```diff
@@ -1,212 +1,302 @@
-   training and validation metrics.
+.. literalinclude:: ../../../examples/tutorials/core_api_pytorch_mnist/model_def_metrics.py
+   :language: python
+   :start-after: # Docs snippet start: modify main loop core context
+   :end-at: main(core_context=core_context)
 
-The complete ``1_metrics.py`` and ``1_metrics.yaml`` listings used in this example can be found in
-the :download:`core_api.tgz </examples/core_api.tgz>` download or in the `Github repository
-<https://github.com/determined-ai/determined/tree/master/examples/tutorials/core_api>`_.
+Step 2.2: Modify the Train Method
+=================================
 
-.. _core-checkpoints:
+Use ``core_context.train`` to report training and validation metrics.
 
-********************
- Report Checkpoints
-********************
+Modify the train() method by adding ``core_context.train.report_training_metrics()``:
 
-By checkpointing periodically during training and reporting those checkpoints to the master, you can
-stop and restart training in two different ways: either by pausing and reactivating training using
-the WebUI, or by clicking the Continue Trial button after the experiment completes.
+.. literalinclude:: ../../../examples/tutorials/core_api_pytorch_mnist/model_def_metrics.py
+   :language: python
+   :start-after: # Docs snippet start: report training metrics
+   :end-before: # Docs snippet end: report training metrics
+   :dedent:
 
-These two types of continuations have different behaviors. While you always want to preserve the
-value you are incrementing (the "model weight"), you do not always want to preserve the batch index.
-When you pause and reactivate you want training to continue from the same batch index, but when
-starting a fresh experiment you want training to start with a fresh batch index. You can save the
-trial ID in the checkpoint and use it to distinguish the two types of continues.
+and ``core_context.train.report_validation_metrics()``:
 
-#. Create a new ``2_checkpoints.py`` training script called by copying the ``1_metrics.py`` script
-   from :ref:`core-metrics`.
+.. literalinclude:: ../../../examples/tutorials/core_api_pytorch_mnist/model_def_metrics.py
+   :language: python
+   :start-after: # Docs snippet start: report validation metrics
+   :end-before: # Docs snippet end: report validation metrics
+   :dedent:
 
-#. Write save and load methods for your model:
+.. note::
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/2_checkpoints.py
-      :language: python
-      :pyobject: save_state
+   Viewing Epoch-Based Metrics in the WebUI
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/2_checkpoints.py
-      :language: python
-      :pyobject: load_state
+   You can modify your code to report epoch-based metrics as shown in this example. When you report
+   epoch-based metrics during training, you'll be able to view epoch-based metric data in the WebUI.
```

Comment:
> you'll be able to view epoch-based metric data in the WebUI

I imagine this claim would be confusing to users because in the Web UI all metrics are listed in terms of batches, not epochs, regardless of how they're reported (I believe)

Reply:
i am leaning towards removing anything about viewing epoch-based metrics in the WebUI from this user guide; i've created a separate "how-to article" (How to View Epoch-Based Metrics in the WebUI). so we're good to leave this out.

Reply:
Good.  Just in general, these entry-level user guides shouldn't be the place to document new features.

It's good to add docs for new features, but not to these user guides.  Find somewhere else, don't hijack the existing tutorials.

Reply:
need to remove code from model_def_metrics.py regarding epoch-based training

`core_context.train.report_training_metrics( 

    steps_completed=(batch_idx + 1) + (epoch - 1) * len(train_loader), 

    metrics={ 

         "train_loss": loss.item(), 

         “epoch”: epoch 

     }, 

) 

 

core_context.train.report_validation_metrics( 

    steps_completed=steps_completed, 

    metrics={ 

      "test_loss": test_loss, 

      “epoch”:epoch 

    }, 

) `

Reply:
sounds good to me

Reply:
thanks for the edits! this is resolved.